packages/adblocker/src/engine/engine.ts

```diff
@@ -697,23 +753,44 @@ export default class FilterEngine extends EventEmitter<
     url,
     hostname,
     domain,
+
+    reference,
```

Comment:
What is `reference`?

Reply:
It's an arbitrary object coming from the user to identify or tag the source of match. This gives an ability to keep tracking of the matches on fly.